completed service heathcare

amanmadaan30 opened the pull request.

amanmadaan30 committed:
completed service heathcare

amanmadaan30 commented:
completed writing policies for gcp service healthcare

amanmadaan30 committed:
Merge branch 'dev' into gcp/service/healthcare

amanmadaan30 committed:
updated policies and tested using helpers.rego

Pat-Stuart commented on policies/gcp/healthcare/consent_store/default_consent_ttl/policy.rego:
```diff
@@ -0,0 +1,30 @@
+            "condition": "non-compliant default_consent_ttl",
+            "attribute_path": ["default_consent_ttl"],
+            "values": ["3600s"],  # explicitly blacklist this value
+            "policy_type": "blacklist"
```
You probably want this as a whitelist.
This should be revisited next sem with new helpers function.

amanmadaan30 replied:
updated to whitelist and tested. works fine.

Pat-Stuart commented on policies/gcp/healthcare/consent_store/default_consent_ttl/policy.rego:
This is more compliance than security but I think this is still relevant.

Pat-Stuart commented on policies/gcp/healthcare/consent_store/default_consent_ttl/policy.rego:
```diff
@@ -0,0 +1,30 @@
+    ]
+]
+
+variables := {
```
Remove this.

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/consent_store/enable_consent_create_on_update/policy.rego:
```diff
@@ -0,0 +1,30 @@
+    ]
+]
+
+variables := {
```
Remove this.

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/consent_store/enable_consent_create_on_update/policy.rego:
```diff
@@ -0,0 +1,30 @@
+        {
+            "condition": "non-compliant enable_consent_create_on_update",
+            "attribute_path": ["enable_consent_create_on_update"],
+            "values": ["false"],  # explicitly blacklist false
```
This should probably be whitelist.

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/consent_store/enable_consent_create_on_update/policy.rego:
More compliance but I think still relevant.

Pat-Stuart commented on policies/gcp/healthcare/consent_store/enable_consent_create_on_update/policy.rego:
```diff
@@ -0,0 +1,30 @@
+    "resource_value_name": "enable_consent_create_on_update"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Set this up like the template just for consistency

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/consent_store/vars.rego:
Update this to the new vars template.

amanmadaan30 replied:
updated vars.rego for all policies.

Pat-Stuart commented on policies/gcp/healthcare/dataset/encryption_spec/policy.rego:
```diff
@@ -0,0 +1,30 @@
+        {
+            "condition": "Malformed KMS key name",
+            "attribute_path": ["encryption_spec", 0, "kms_key_name"],
+            "values": ["^projects/[^/]+/locations/[^/]+/keyRings/[^/]+/cryptoKeys/[^/]+$"],
```
This isn't how our pattern whitelist works.

amanmadaan30 replied:
updated and tested

Pat-Stuart commented on policies/gcp/healthcare/dataset/encryption_spec/policy.rego:
```diff
@@ -0,0 +1,30 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/dataset/encryption_spec/policy.rego:
```diff
@@ -0,0 +1,30 @@
+    "resource_value_name": "name"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Set this up like the template just for consistency

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/dataset/time_zone/policy.rego:
Not security related

amanmadaan30 replied:
I believe a mismatched time zone can lead to security issues. For example, a situation can arise when systems interacting with the dataset operate in different or conflicting time zones, security alerts, job scheduling, backups, and automated rotations may get misaligned. I am still open for discussions. Let me know if this point is valid, otherwise I can remove this policy.

Pat-Stuart replied:
This would depend on how the data is being stored, often there is an explicit timestamp however without this knowledge I largely agree.

It makes more sense to whitelist a specific value however

amanmadaan30 replied:
pretty much valid in whitelisting, updated and tested.

Pat-Stuart commented on policies/gcp/healthcare/dataset/vars.rego:
Update this to the new vars template

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/default_search_handling_strict/policy.rego:
Not security related.

amanmadaan30 replied:
sure, removed.

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_referential_integrity/policy.rego:
```diff
@@ -0,0 +1,29 @@
+            "condition": "non-compliant disable_referential_integrity",
+            "attribute_path": ["disable_referential_integrity"],
+            "values": [true],
+            "policy_type": "blacklist"
```
This should be a whitelist

amanmadaan30 replied:
updated and tested. works fine.

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_referential_integrity/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_referential_integrity/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    "resource_value_name": "disable_referential_integrity"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Update like template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_referential_integrity/policy.rego:
Arguably not security related but related to data integrity.

amanmadaan30 replied:
I consider this to be security related as it could lead to tampering risk as an attacker could intentionally reference non-existent or forged resource IDs, leading to misleading or manipulated records. Open for discussion and can remove if still required.

Pat-Stuart replied:
Fair enough, I agree that it should be enabled in any case

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_resource_versioning/policy.rego:
This is more to do with compliance and audit but I think it probably counts

amanmadaan30 replied:
yes, this can be helpful in some adverse situations where an attacker can update consent record or medical notes without versioning.

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_resource_versioning/policy.rego:
```diff
@@ -0,0 +1,29 @@
+            "condition": "non-compliant disable_resource_versioning",
+            "attribute_path": ["disable_resource_versioning"],
+            "values": [true],
+            "policy_type": "blacklist"
```
You probably want to whitelist false

amanmadaan30 replied:
updated and tested. works fine.

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_resource_versioning/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/disable_resource_versioning/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    "resource_value_name": "disable_resource_versioning"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Change to conform with template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_history_import/policy.rego:
More to do with data compliance and integrity

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_history_import/policy.rego:
```diff
@@ -0,0 +1,29 @@
+            "condition": "non-compliant enable_history_import",
+            "attribute_path": ["enable_history_import"],
+            "values": [true],
+            "policy_type": "blacklist"
```
You should whitelist false

amanmadaan30 replied:
policy removed as not related to security

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_history_import/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_history_import/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    "resource_value_name": "enable_history_import"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Change to conform with template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_update_create/policy.rego:
```diff
@@ -0,0 +1,29 @@
+            "condition": "non-compliant enable_update_create",
+            "attribute_path": ["enable_update_create"],
+            "values": [true],
+            "policy_type": "blacklist"
```
Should whitelist false

amanmadaan30 replied:
policy removed as not related to security

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_update_create/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_update_create/policy.rego:
```diff
@@ -0,0 +1,29 @@
+    "resource_value_name": "enable_update_create"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Change to conform with template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/enable_update_create/policy.rego:
Compliance and data integrity related

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/fhir_store/vars.rego:
Change to new vars template

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/labels/policy.rego:
```diff
@@ -0,0 +1,32 @@
+        {
+            "condition": "non-compliant label keys",
+            "attribute_path": ["labels"],
+            "values": ["patient_id", "ssn", "user_id", "confidential", "pii"],
```
Include Australian PII such as medicare

amanmadaan30 replied:
included and tested

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/labels/policy.rego:
```diff
@@ -0,0 +1,32 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/labels/policy.rego:
```diff
@@ -0,0 +1,32 @@
+    "resource_value_name": "labels"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Change to conform with template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/notification_configs/policy.rego:
Probably not security related but open to justification

amanmadaan30 replied:
I did some more research, we can probably remove this.

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/vars.rego:
Change to new vars template

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/parser_config/policy.rego:
```diff
@@ -0,0 +1,34 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/parser_config/policy.rego:
```diff
@@ -0,0 +1,34 @@
+            ]
+        },
+        {
+            "condition": "non-compliant parser_config settings",
```
These need to be separated out

amanmadaan30 replied:
policy removed

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/parser_config/policy.rego:
Arguably not security related

amanmadaan30 replied:
after some more research, i realised that this is more related to data and compliance. We can remove this.

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/parser_config/policy.rego:
```diff
@@ -0,0 +1,34 @@
+        },
+        {
+            "condition": "non-compliant parser_config settings",
+            "attribute_path": ["parser_config.allow_null_header", "parser_config.segment_terminator", "parser_config.version"],
```
Attribute paths are incorrect

amanmadaan30 replied:
policy removed

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/notification_configs/policy.rego:
```diff
@@ -0,0 +1,33 @@
+        },
+        {
+            "condition": "non-compliant notification config",
+            "attribute_path": ["notification_configs[*].pubsub_topic", "notification_configs[*].filter"],
```
Attribute path is incorrect

amanmadaan30 replied:
policy removed as was not related to security.

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/notification_configs/policy.rego:
```diff
@@ -0,0 +1,33 @@
+    ]
+]
+
+variables := {
```
Remove this

amanmadaan30 replied:
removed

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/notification_configs/policy.rego:
```diff
@@ -0,0 +1,33 @@
+    "resource_value_name": "notification_configs"
+}
+
+summary := helpers.get_multi_summary(scenarios_list, variables)
```
Change to conform with template

amanmadaan30 replied:
updated

Pat-Stuart commented on policies/gcp/healthcare/hl7_v2_store/labels/policy.rego:
Requires testing as I think labels wont be accessed correctly here

amanmadaan30 replied:
tested the policy
![ss](https://github.com/user-attachments/assets/f001bcf0-4832-4feb-abea-08dc6f6e5435)

Pat-Stuart replied:
The test shows that the policy fails to detect your NC resource. 

You are only accessing this at the object/block level. There may not currently be a way to do what you are trying to achieve, something to look into with new helpers functions

amanmadaan30 replied:
understood. tried some debugging but could not find a way to access resource using helpers. there might be some changes in helpers required or some custom message in policy to access that.

Pat-Stuart reviewed (changes requested):
Folder structure good. Indentation good.

A bit of work to fix up the policies and some are not security related.

Please update the resource level vars and remove individual policy vars.
Overall good job, thanks.

amanmadaan30 committed:
updated changes as per p/r comments

amanmadaan30 commented:
carefully gone through all the comments and updated changes wherever required. let me know if anything else is required. thanks for detailed code review.

Pat-Stuart commented:
Thanks for addressing the feedback.
Apologies but something I did miss is to change scenario_list to conditions in each policy. This will just keep it consistent with all other services for automation purposes.
Largely looks good now, just have a look at labels. Might not currently be possible

amanmadaan30 committed:
addressed p/r comments

amanmadaan30 commented:
updated scenario_list to conditions. addressed rest of the feedbacks. for labels, I could not find a way for now. Please suggest if there can be a way to address that.

Pat-Stuart commented:
> updated scenario_list to conditions. addressed rest of the feedbacks. for labels, I could not find a way for now. Please suggest if there can be a way to address that.

After discussion with Yotam we are going to be archiving policies for labels for the time being. 
Thanks for your time addressing feedback, policies look good now.

Pat-Stuart commented:
dataset.encryption_spec is probably broken and labels should probably be removed. Closing for minor changes
